prime-router/src/main/kotlin/Mappers.kt

```diff
@@ -122,6 +139,71 @@ class UseMapper : Mapper {
     }
 }
 
+/**
+ * Use a string value found in the Sender's setting object as the value for this element.
+ * Example call
+ *   - name: sender_id
+ *      cardinality: ONE
+ *      mapper: useSenderSetting(fullName)
+ *      csvFields: [{ name: senderId}]
+ *
+ * As of this writing mappers are called in three places:
+ *  - during initial read (in [Element.processValue])
+ *  - during creation of internal data (in [Report.buildColumnPass2])
+ *  - during creation of outgoing data (in, eg, [Hl7Serializer.setComponentForTable])
+ *  ONLY the first of those has access to Sender info. However, the mapper may be called in
+ *  any or all of those three places. Therefore, its incumbent upon the
+ *  writer of any mapper to ensure it works without failure if the Sender obj is null.
+ *
+ *  Notes:
+ *  1. If you use mapperOverridesValue with this, you will get unexpected results.
+ *  2. [UseSenderSettingMapper] always returns the toString() value regardless of the field type.
+ *  3. This does not work with commandline ./prime, because the CLI knows nothing about settings.
+ */
+class UseSenderSettingMapper : Mapper {
+    override val name = "useSenderSetting"
+
+    override fun valueNames(element: Element, args: List<String>): List<String> {
+        if (args.size != 1) {
+            error("Schema Error for ${element.name}: useSenderSetting expects a single argument")
+        }
+        // The arg is the name of the field in the Sender settings to extract, not a field in the data.
+        return emptyList()
+    }
+
+    override fun apply(
+        element: Element,
+        args: List<String>,
+        values: List<ElementAndValue>,
+        sender: Sender?
+    ): ElementResult {
+        val valueToUse =
+            when {
+                sender == null -> null // will null out existing value if you use mapperOverridesValue
+                args.size != 1 -> error("Schema Error for ${element.name}: useSenderSetting expects a single argument")
+                else -> {
+                    try {
+                        val senderProperty = Sender::class.memberProperties.first {
+                            it.name == args[0]
+                        }
+                        senderProperty.get(sender).toString()
```

Comment:
If the sender property exists, but is null, won't the call `toString` cause an error, but not the error you track below, which could lead to confusion.